exercises/main.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+func bisectSearch(s []int, x int) int {
+	start, end := 0, len(s)-1
+	for start <= end {
+		mid := (start + end) / 2
+		if s[mid] == x {
+			return mid
+		} else if s[mid] < x {
+			start = mid + 1
+		} else {
+			end = mid - 1
+		}
+	}
+	return start
+}
+
+func insertionSortBisectSearch(s []int) {
+	for i := 1; i < len(s); i++ {
+		x := bisectSearch(s[:i+1], s[i])
+		for j := i; j > x; j-- {
+			s[j], s[j-1] = s[j-1], s[j]
```

Comment:
I think you miss smth here. You will miss s[i]-th value

Reply:
sorry, my fault - you swap elements, which means that you don't lose it. things look good